File: src/main/java/voidchess/image/Paintable.java
package voidchess.image;

import java.awt.*;
import java.awt.image.ImageObserver;

/**
 * @author stephan
 */
public class Paintable {
    private final ImageObserver observer;
    private final Image image;

    public Paintable(ImageObserver observer, Image image) {
        this.observer = observer;
        this.image = image;
    }

    public void paintOn(Graphics g, int x_pos, int y_pos, int areaSize) {
        g.drawImage(image, x_pos, y_pos, areaSize, areaSize, observer);
    }
}

File: src/main/kotlin/voidchess/image/Paintable.kt
package voidchess.image

import java.awt.*
import java.awt.image.ImageObserver


class Paintable(private val observer: ImageObserver, private val image: Image) {

    fun paintOn(g: Graphics, x_pos: Int, y_pos: Int, areaSize: Int) {
        g.drawImage(image, x_pos, y_pos, areaSize, areaSize, observer)
    }
}
